demos/python_demos/colorization/colorization.py

```diff
@@ -0,0 +1,138 @@
+    in_args.add_argument("-m", "--model", help="Required. Path to .xml file with pre-trained model",
+                         required=True, type=str)
+    in_args.add_argument("-d", "--device",
+                         help="Optional. Specify target device for infer: CPU, GPU, FPGA, HDDL or MYRIAD. "
+                         "Default: CPU",
+                         default="CPU", type=str)
+    in_args.add_argument("-i", "--input", help="Optional. Path to a test video file.", type=str)
+    in_args.add_argument("-n", "--no_show", help="Optional. Disable display of results on screen.",
+                         action='store_true', default=False)
+    in_args.add_argument("-v", "--verbose", help="Optional. Enable display of processing logs on screen.",
+                         action='store_true', default=False)
+    return parser
+
+
+def get_parameters(xml_path):
+    root = ET.parse(xml_path).getroot()
+    pattern = r'data\[([0-9]*\.*[0-9]*)\]'
+    scale_data = root.findall("./meta_data/cli_parameters/scale_values")[0]
+    mean_data = root.findall("./meta_data/cli_parameters/mean_values")[0]
+    scale_const = float(re.match(pattern, scale_data.attrib['value']).group(1))
+    mean_const = float(re.match(pattern, mean_data.attrib['value']).group(1))
+    return scale_const, mean_const
+
+
+def get_model_files(args_in):
+    config_xml_in = args_in.model
+    model_path = os.path.splitext(config_xml_in)[0]
+    weights_bin_in = model_path + ".bin"
+    level_up = os.path.split(os.path.split(model_path)[0])[0]
+    coeffs_in = os.path.join(level_up, os.path.split(level_up)[1]) + ".npy"
+    return config_xml_in, weights_bin_in, coeffs_in
+
+
+if __name__ == '__main__':
+    args = build_arg().parse_args()
+    config_xml, weights_bin, coeffs = get_model_files(args)
+
+    # scale and mean are stored in the source caffemodel
+    scale, mean = get_parameters(config_xml)
+
+    log.basicConfig(format="[ %(levelname)s ] %(message)s",
+                    level=log.INFO if not args.verbose else log.DEBUG, stream=sys.stdout)
+
+    log.debug("Load network")
+    load_net = IENetwork(model=config_xml, weights=weights_bin)
+    exec_net = IECore().load_network(network=load_net, device_name=args.device)
+
+    assert len(load_net.inputs) == 1, "Number of inputs does not match network inputs"
+    input_blob = next(iter(load_net.inputs))
+    input_shape = load_net.inputs[input_blob].shape
+    assert input_shape == [1, 1, 224, 224], "Shape of inputs does not match network shape inputs"
+
+    assert len(load_net.outputs) == 1, "Number of outputs does not match network outputs"
+    output_blob = next(iter(load_net.outputs))
+    output_shape = load_net.outputs[output_blob].shape
+    assert output_shape == [1, 313, 56, 56], "Shape of outputs does not match network shape outputs"
+
+    n, c, h_in, w_in = input_shape
+
+    cap = cv.VideoCapture(args.input if args.input else 0)
```

Comment:
why will not be optional?

Reply:
`input_source = args.input` won't rise an exception. You should have cast `args.input` to int here, and if it fails, just pass `args.input` unchanged.

Reply:
@Wovchena done)

Reply:
```
    if not os.path.isdir(input_source):
        assert "{} not exist".format(input_source)
```
I don't get this check, but any way, you better check that `cap.isOpened()` instead of path existence, because VideoCapture can accept other inputs like `img_%02d.jpg`.

The same is for the help message for `-i`: it is up to OpenCV which inputs are supported. Please, reformulate as `Required. Input to process.`. See https://github.com/opencv/open_model_zoo/blob/develop/CONTRIBUTING.md#demo